File: BusTripManagement.Models/RouteStop.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BusTripManagement.Models
{
    public class RouteStop
    {
        public int RouteId { get; set; }
        public int StopId { get; set; }
        public int Sequence { get; set; }

        public string Name { get; set; } = "";
        public double Lat { get; set; }
        public double Lng { get; set; }
    }
}

File: flutter_app/lib/models/routeStop.dart
class RouteStop {
  final int routeId;
  final int stopId;
  final int sequence;
  final String name;
  final double lat;
  final double lng;

  RouteStop({
    required this.routeId,
    required this.stopId,
    required this.sequence,
    required this.name,
    required this.lat,
    required this.lng,
  });

  factory RouteStop.fromJson(Map<String, dynamic> json) {
    return RouteStop(
      routeId: json["routeId"] ?? 0,
      stopId: json["stopId"] ?? 0,
      sequence: json["sequence"] ?? 0,
      lat: (json["lat"] as num?)?.toDouble() ?? 0.0,
      lng: (json["lng"] as num?)?.toDouble() ?? 0.0,
      name: json["name"] ?? "Unknown",
    );
  }
}
